sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientTest.java

```diff
@@ -408,4 +414,150 @@ public void analyseSentimentForBatchInputShowStatistics() {
+            client.detectLanguage("This is a test English Text").getDetectedLanguages());
+    }
+
+    /**
+     * Test client builder with invalid subscription key
+     */
+    @Test
+    public void invalidKey() {
+        // Arrange
+        final TextAnalyticsClient client = createClientBuilder(getEndpoint(),
+            new TextAnalyticsSubscriptionKeyCredential(INVALID_KEY)).buildClient();
+
+        // Action and Assert
+        assertThrows(HttpResponseException.class, () -> client.detectLanguage("This is a test English Text"));
+    }
+
+    /**
+     * Test client with valid subscription key but update to invalid key and make call to server.
+     */
+    @Test
+    public void updateToInvalidKey() {
+        // Arrange
+        final TextAnalyticsSubscriptionKeyCredential credential =
+            new TextAnalyticsSubscriptionKeyCredential(getSubscriptionKey());
+
+        final TextAnalyticsClient client = createClientBuilder(getEndpoint(), credential).buildClient();
+
+        // Update to invalid key
+        credential.updateCredential(INVALID_KEY);
+
+        // Action and Assert
+        assertThrows(HttpResponseException.class, () -> client.detectLanguage("This is a test English Text"));
+    }
+
+    /**
+     * Test client with invalid subscription key but update to valid key and make call to server.
+     */
+    @Test
+    public void updateToValidKey() {
+        // Arrange
+        final TextAnalyticsSubscriptionKeyCredential credential =
+            new TextAnalyticsSubscriptionKeyCredential(INVALID_KEY);
+
+        final TextAnalyticsClient client = createClientBuilder(getEndpoint(), credential).buildClient();
+
+        // Update to valid key
+        credential.updateCredential(getSubscriptionKey());
+
+        // Action and Assert
+        validateDetectedLanguages(Arrays.asList(new DetectedLanguage("English", "en", 1.0)),
+            client.detectLanguage("This is a test English Text").getDetectedLanguages());
+    }
+
+    /**
+     * Test for missing endpoint
+     */
+    @Test
+    public void missingEndpoint() {
+        assertThrows(NullPointerException.class, () -> {
+            final TextAnalyticsClientBuilder builder = new TextAnalyticsClientBuilder();
+            builder.buildClient();
+        });
+    }
+
+    /**
+     * Test for null subscription key
+     */
+    @Test
+    public void nullSubscriptionKey() {
+        assertThrows(NullPointerException.class, () -> {
+            final TextAnalyticsClientBuilder builder = new TextAnalyticsClientBuilder();
+            builder.endpoint(getEndpoint()).subscriptionKey(null).buildClient();
+        });
+    }
+
+    /**
+     * Test for null AAD credential
+     */
+    @Test
+    public void nullAADCredential() {
+        assertThrows(NullPointerException.class, () -> {
+            final TextAnalyticsClientBuilder builder = new TextAnalyticsClientBuilder();
+            builder.endpoint(getEndpoint()).credential(null).buildClient();
```

Comment:
Generally, you try to limit the scope of your assertion. How do you know it threw in credential(null) and not buildclient()?

This goes for all the other tests that do the same.

```java
// arrange
final TextAnalyticsClientBuilder builder = new TextAnalyticsClientBuilder();

// Act & Assert
assertThrows(NullPointerException.class, () -> builder.credential(null));
```

Reply:
Make sense. I will update it.